src/chart.zoom.js

```diff
@@ -376,23 +385,42 @@ var zoomPlugin = {
 			chartInstance.zoom._mouseUpHandler = function(event){
 				if (chartInstance.zoom._dragZoomStart) {
 					var chartArea = chartInstance.chartArea;
+					var xAxis = getXAxis(chartInstance);
 					var yAxis = getYAxis(chartInstance);
 					var beginPoint = chartInstance.zoom._dragZoomStart;
-					var offsetX = beginPoint.target.getBoundingClientRect().left;
-					var startX = Math.min(beginPoint.clientX, event.clientX) - offsetX;
-					var endX = Math.max(beginPoint.clientX, event.clientX) - offsetX;
-					var dragDistance = endX - startX;
-					var chartDistance = chartArea.right - chartArea.left;
-					var zoom = 1 + ((chartDistance - dragDistance) / chartDistance );
+					var startX = xAxis.left;
+					var endX = xAxis.right;
+					var startY = yAxis.top;
+					var endY = yAxis.bottom;
+
+					if (directionEnabled(options.zoom.mode, 'x')) {
+						var offsetX = beginPoint.target.getBoundingClientRect().left;
+						startX = Math.min(beginPoint.clientX, event.clientX) - offsetX;
+						endX = Math.max(beginPoint.clientX, event.clientX) - offsetX;
+					}
+
+					if (directionEnabled(options.zoom.mode, 'y')) {
+						var offsetY = beginPoint.target.getBoundingClientRect().top;
+						startY = Math.min(beginPoint.clientY, event.clientY) - offsetY;
+						endY = Math.max(beginPoint.clientY, event.clientY) - offsetY;
+					}
+
+					var dragDistanceX = endX - startX;
+					var chartDistanceX = chartArea.right - chartArea.left;
+					var zoomX = 1 + ((chartDistanceX - dragDistanceX) / chartDistanceX );
```

Comment:
Please remove the extra space before `);` (same below). After you rebase against master, you can run `gulp lint` which will automatically check the PR for style issues